plugins/module_utils/azure_rm_common_ext.py

```diff
@@ -213,44 +257,54 @@ def default_compare(self, modifiers, new, old, path, result):
             else:
                 return True
 
-    def update_identities(self, curr_identity):
-        curr_identity = curr_identity or dict()
-        changed = False
-        current_managed_type = curr_identity.get('type', 'None')
-        current_managed_identities = set(curr_identity.get('user_assigned_identities', {}).keys())
-        param_identity = self.module.params.get('identity', {})
-        param_identities = set(param_identity.get('user_assigned_identities', {}).get('id', []))
-        external_identities = param_identities
+    def update_single_managed_identity(self, curr_identity, new_identity):
+        # Converting from single_managed_identity_spec to managed_identity_spec
+        new_identity_converted = {
+            'type': new_identity.get('type', 'None'),
+        }
+        user_assigned_identity = new_identity.get('user_assigned_identity', None)
+        if user_assigned_identity is not None:
+            new_identity_converted['user_assigned_identities'] = {
+                'id': [user_assigned_identity]
+            }
+        return self.update_managed_identity(curr_identity=curr_identity, new_identity=new_identity_converted, allow_identities_append=False)
 
+    def update_managed_identity(self, curr_identity=None, new_identity=None, allow_identities_append=True):
+        curr_identity = curr_identity or dict()
+        new_identity = new_identity or self.module.params.get('identity', {})
```

Comment:
```
-        new_identity = new_identity or self.module.params.get('identity', {})
+        new_identity = new_identity or self.module.params.get('identity', {}) or dict()
```

Reply:
Fixed

Reply:
I also added comment here to remove `self.module.params.get('identity', {})` after all modules are aligned to this change